File: py-basics/src/lectures/classes/song.py
# 7 kyu | What a "Classy" Song
# https://www.codewars.com/kata/6089c7992df556001253ba7d/python

class Song:
    def __init__(self, title, artist):
        self.title = title
        self.artist = artist
        self.listeners = set()

    def how_many(self, people):
        old_length = len(self.listeners)
        for person in people:
            self.listeners.add(person.lower())
        return len(self.listeners) - old_length


def test(expected, actual):
    checkmark = '✔️' if expected == actual else '❌'
    print(f"{checkmark} expected -> actual: {str(expected)} -> {str(actual)}")


mount_moose = Song('Mount Moose', 'The Snazzy Moose')
test('Mount Moose', mount_moose.title)
test('The Snazzy Moose', mount_moose.artist)
test(5, mount_moose.how_many(['John', 'Fred', 'Bob', 'Carl', 'RyAn']))
test(2, mount_moose.how_many(['JoHn', 'Luke', 'AmAndA']))
test(2, mount_moose.how_many(['Amanda', 'CalEb', 'CarL', 'Furgus']))
test(1, mount_moose.how_many(['JOHN', 'FRED', 'BOB', 'CARL', 'RYAN', 'KATE']))
test(1, mount_moose.how_many(['Jack', 'jacK']))

File: js-basics/src/lectures/classes/song.js
/**
 * 7 kyu | What a "Classy" Song
 * https://www.codewars.com/kata/6089c7992df556001253ba7d/javascript
 */

class Song {
    constructor(title, artist) {
        // TODO
    }
    howMany(people) {
        return [];
    }
}

function test(actual, expected) {
    let checkmark = expected == actual ? '✔️' : '❌';
    console.log(`${checkmark} expected -> actual: ${new String(expected)} -> ${new String(actual)}`);
}

let mountMoose = new Song('Mount Moose', 'The Snazzy Moose');
test('Mount Moose', mountMoose.title);
test('The Snazzy Moose', mountMoose.artist);
test(5, mountMoose.howMany(['John', 'Fred', 'Bob', 'Carl', 'RyAn']));
test(2, mountMoose.howMany(['JoHn', 'Luke', 'AmAndA']));
test(2, mountMoose.howMany(['Amanda', 'CalEb', 'CarL', 'Furgus']));
test(1, mountMoose.howMany(['JOHN', 'FRED', 'BOB', 'CARL', 'RYAN', 'KATE']));
test(1, mountMoose.howMany(['Jack', 'jacK']));
